doc/guides/maintaining-c-ares.md

```diff
@@ -0,0 +1,63 @@
+# Maintaining c-ares
+
+Updates to the c-ares dependency involve the following steps:
+1. Downloading the source archive for the new version.
+1. Unpacking the source in a temporary workspace directory.
+1. Removing the `test` directory (to save disk space).
+1. Copying over the existing `.gitignore`, pre-generated `config` directory and
+   `cares.gyp` files.
+1. Replacing the existing `deps/cares` with the workspace directory.
+1. Modifying the `cares.gyp` file for file additions/deletions.
+1. Rebuilding the main Node.js `LICENSE`.
```

Comment:
Not sure if this needs to be formatted @Trott.

Reply:
I added suggestions from the output of `node tools/lint-md/lint-md.src.mjs --format doc/guides/maintaining-c-ares.md`. (The comma suggestion is not from the tool, of course. But the others all are.)

Reply:
Thank you @Trott!